aries_cloudagent/admin/server.py

```diff
@@ -285,7 +301,58 @@ async def collect_stats(request, handler):
 
             middlewares.append(collect_stats)
 
+        if self.context.settings.get("multitenant.enabled"):
+
+            @web.middleware
+            async def set_multitenant_wallet(request, handler):
+                authorization_header = request.headers.get("Authorization")
+
+                # MTODO better auth. Now if no authorization header  is present everything
+                # can be done on base wallet. We want to limit this to only subwallet
+                # So we need to check if it's a multitenancy api path and otherwise reject
+                if is_unprotected_path(request.path) or not authorization_header:
+                    return await handler(request)
+
+                context = request["context"]
+                jwt_secret = context.settings.get("multitenant.jwt_secret")
+
+                # Extract token and check if Bearer is present
+                bearer, _, token = authorization_header.partition(" ")
+                if bearer != "Bearer":
+                    raise web.HTTPUnauthorized(reason="Invalid header structure")
+
+                try:
+                    token_body = jwt.decode(token, jwt_secret)
+
+                    # Get the wallet associated with the subwallet
+                    wallet_id = token_body["wallet_id"]
+                    wallet_record = cast(
+                        WalletRecord,
+                        await WalletRecord.retrieve_by_id(context, wallet_id),
+                    )
+
+                    context.settings = context.settings.extend(
+                        wallet_record.get_config_as_settings()
+                    )
+
+                    if wallet_record.key_management_mode == WalletRecord.MODE_UNMANAGED:
+                        context.settings["wallet.key"] = token_body["wallet_key"]
+                # MTODO: catch specific exceptions
+                except Exception:
+                    raise web.HTTPUnauthorized()
+
+                # set the updated request context
+                request["context"] = context
+
+                return await handler(request)
+
+            middlewares.append(set_multitenant_wallet)
+
```

Comment:
If multitenancy is enabled, we check for the `Authorization` header. If it is available we check for validity and update the request context settings. Whenever the wallet is injected, the subwallets config will be used from now on